src/quest/manager.py

```diff
@@ -41,10 +47,13 @@ def __init__(self, namespace: str, storage: BlobStorage, create_history: History
         self._workflows: dict[str, Historian] = {}
         self._workflow_tasks: dict[str, asyncio.Task] = {}
         self._alias_dictionary = {}
-
+        self._custom_signal_handler = None
 
     async def __aenter__(self) -> 'WorkflowManager':
         """Load the workflows and get them running again"""
+
+        signal.signal(signal.SIGINT, self.dispatch_signal_handler)
+
```

Comment:
grab the existing signal handler and remember it.

in `__aexit__`, put it back.

In our handler, call the existing handler (if any) after calling ours.